app/models/publisher.rb

```diff
@@ -19,6 +19,27 @@ def publish
     end
   end
 
+  def discard_draft
+    begin
+      ActiveRecord::Base.transaction do
+        publishing_api.discard_draft(content_model.content_id)
+
+        latest_published_edition = content_model.editions.published.last
+        if latest_published_edition.present?
+          content_model
+            .editions
+            .where("created_at > ?", latest_published_edition.created_at)
+            .destroy_all
+        else
+          content_model.destroy!
```

Comment:
I reckon we just validate against it rather than remove it from the topic. Because the relation is in a blob of JSON we can't use the `dependent: :destroy` niceness.